### ISSUE TITLE
Requirements for unfreezing AI law changes

### ISSUE THREAD

**Issue by slarticodefast** [space-wizards/space-station-14#31924]
We had a huge influx of AI law addition PRs, but AI is not ready for that yet until we address some basics:
- [ ] Decide how we want laws and law uploads to work in general and which ones to use as a default. This will need an accepted design doc.
- [ ] Implement freeform law upload. Ideally this should allow to also allow to select from presets. It should be properly admin logged.
- [ ] Make the AI destructible so the crew can fight against it in case it malfunctions.
- [ ] Implement turrets for the AI to allow it to self-defend. These could be based on ID access and the AI should be able to override them.

The following law proposals will be compiled into one PR and discussed with the game admins once these conditions are fulfilled:
- #31709
- #31901
- #31913
- #31855
- #31899

**Comment by t0wnshark** [comment]
Question - I wrote some laws on my own in #31913 that aren't specifically from any codebase or fork. If they eventually get added, will I be credited for them? I'm a little disappointed that my PR was immediately closed because I spent more than six hours writing and rewriting all of the YAML and text descriptions, and a good chunk of that time was coming up with the original lawsets.

**Comment by Psychpsyo** [comment]
I get the point here, but judging from the one PR, the AI is currently not affected by ion storms, making some of these a borg-only change. (and technically admemes)
Are ion-stormed borgs also slated for re-evaluation or why were those frozen?

(yes, this is a thinly veiled "reopen asimoth pls cause it doesn't mention or affect the AI in any way")

**Comment by slarticodefast** [comment]
> Question - I wrote some laws on my own in https://github.com/space-wizards/space-station-14/pull/31913 that aren't specifically from any codebase or fork. If they eventually get added, will I be credited for them? I'm a little disappointed that my PR was immediately closed because I spent more than six hours writing and rewriting all of the YAML and text descriptions, and a good chunk of that time was coming up with the original lawsets.

Yes, we will make sure all attributions are properly credited when they get included. The main reason for compiling them into one PR is so we can coordinate the discussion with the admins better.

>I get the point here, but judging from the one PR, the AI is currently not affected by ion storms, making some of these a borg-only change. (and technically admemes)
Are ion-stormed borgs also slated for re-evaluation or why were those frozen?

Not yet, but that will likely be the case in the future, either as an ion storm or as a malfunctioning AI gamemode. However, even adding laws to borgs is relevant to future AI design, that we haven't decided on yet and we want to avoid having to remove them again later. An examples for this would be a potential 0th law ordering borgs to follow the AI's orders or the AI being able to control borgs. We also will have to think about how laws are to be interpreted as a whole concerning server rules. Are they more roleplay guidelines, or part of your job description and when is violating them a reason for a ban? Some lawsets are also designed to be independent of the law priority we currently have as a part of our silicon rules, while others are not.

> this is a thinly veiled "reopen asimoth pls

We got quite a influx of similar PRs and without the freeze there will be likely many more. Each one of these has gotten an separate internal discussion thread with over 20 maintainers and admins leaving hundreds of comments. The consensus was that we need to agree on a general direction for AI first, which will allow us to make easier decisions about smaller additions like lawsets.

**Comment by t0wnshark** [comment]
> Yes, we will make sure all attributions are properly credited when they get included. The main reason for compiling them into one PR is so we can coordinate the discussion with the admins better.

Great! Just wanted to make sure that my hard work wasn't going to go to a COMPLETE waste.

**Comment by HarrisonGreenlee** [comment]
This might seem a bit odd, but it would be really interesting to make the Lawyer responsible for drafting freeform AI laws, and maybe even managing the AI laws in general. From a logical perspective it would absolutely make sense that Centcom would want to entrust this responsibility to somebody who has experience drafting robust contracts, and this sort of thing is already (somewhat) happening in the real world as well.

From a gameplay perspective, this would also make sense as the lawyer role tends to have too much free time when security doesn't feel like interacting with space law. It could be really cool to give Laywer players the option to interact with the station in a meaningful way during particularly boring rounds instead of just waiting around at the brig. 

It also gives lawyers a tiny bit of leverage over security which can be useful to make security actually care about space law without having to go to annoy the heads of the station every time. And of course, if they abuse the freeform law system they will quickly be demoted, imprisoned, or sued for causing damage to the station and its crew.

If this seems interesting I can look into creating a design doc, just let me know.

**Comment by Psychpsyo** [comment]
> It also gives lawyers a tiny bit of leverage over security which can be useful to make security actually care about space law without having to go to annoy the heads of the station every time. And of course, if they abuse the freeform law system they will quickly be demoted, imprisoned, or sued for causing damage to the station and its crew.

"If you're not letting me prosecute, I'll make the AI put *you* into jail!"

**Comment by JORJ949** [comment]
As someone who has played a LOT of AI on SS13's goonstation I really like how laws are managed there as it is unique to them (and as such would probably need licensing porting) in which they use a server rack with each individual law taking up a slot, this allows for far more intricate lawsets and precedences. Additional law racks can be built to create your own borg army custom laws. The section on their wiki for this can be found [here](https://wiki.ss13.co/AI_Laws#Law_Management).

Syndicate cyborgs are also very different from emagged, where they get a syndicate lawset (like nukie borgs do here) and emagged borgs (which can be done with cover locked) become lawless. Syndicate cyborg creation is exclusive to the traitor roboticist via either a syndicate frame or a conversion chamber.

At present moment I really do not think I will play nearly as much AI on ss14 due to the sheer rigidity of uploadable laws, which is a shame because AI law stuff is what got me really into space station in the first place.

**Comment by ThanosDeGraf** [comment]
> - Make the AI destructible so the crew can fight against it in case it malfunctions.

This point has been cleared!

**Comment by jamilnielsen** [comment]
i know this is not the place for discussions, but i will note that the most important part before we start opening the flood gates is dealing with users wanting with glee force weird and absurd laws on the AI, i saw this quite abit on a ss13 server were people could either A: with few to no limitations get any lawset uploaded to server or B: just do it free-form in round.
it is my belief that AI is better left largely as-is and making any non-default law change a traitor only thing, i have seen RDs or captains with glee in there eye wanting to change AI laws for the sake of fun and that's already pushing it.
i cannot stretch this enough, especially as this easily becomes an ADMIN issue, it needs to be clearly illegal to modify the AI in any way except for restoring default.

i know it would be super fun for AI to do all kinds of wacky things, but they MUST be locked behind traitor to limit abuse.
for this reason i hope this point is taken into account for design docs.

**Comment by SolventMercury** [comment]
> - Decide how we want laws and law uploads to work in general and which ones to use as a default. This will need an accepted design doc.  
  
Just opened a pull for a doc to address this point.
https://github.com/space-wizards/docs/pull/609
